drivers/hmis_supplemental/app/models/hmis_supplemental/data_set.rb

```diff
@@ -0,0 +1,81 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HmisSupplemental
+  class DataSet < GrdaWarehouseBase
+    self.table_name = 'hmis_supplemental_data_sets'
+    has_paper_trail
+
+    belongs_to :remote_credential, class_name: 'GrdaWarehouse::RemoteCredentials::S3'
+    accepts_nested_attributes_for :remote_credential
+
+    belongs_to :data_source, class_name: 'GrdaWarehouse::DataSource'
+    validates :owner_type, inclusion: { in: ['enrollment', 'client'] }
+    has_many :field_values,
+             class_name: 'HmisSupplemental::FieldValue',
+             foreign_key: 'data_set_id',
+             dependent: :restrict_with_exception
+    validates :name, presence: true
+    validates :object_key, presence: true
+    has_many :group_viewable_entities,
+             class_name: 'GrdaWarehouse::GroupViewableEntity',
+             as: :entity,
+             dependent: :destroy
+
+    scope :viewable_by, ->(user) do
```

Comment:
something like this may already exist. Happy to dry up if it's there's another implementation

Reply:
We've implemented this in each model that is represented by a collection, but I believe the logic is different (by design) for each.

Reply:
Since we only have one permission, it's probably not necessary, but this might want to take a permission argument for future expansion.

```suggestion
    scope :viewable_by, ->(user, permission: :can_view_supplemental_client_data) do
```

Reply:
not sure, maybe the permission used to determine visibility should stay as internal implementation; not something that the caller should know about.

Reply:
I guess I see these scopes more as `accessible_by_within_context(user, context)` where context is a permission, but let's leave it as is for now, perhaps the policy provider will make it more clear what scopes we'll need.